### PR TITLE
Add HuffPost

### PR DESCRIPTION
**Pull request by KhalidAlt**

### CODE REVIEW

**Comment by jon-tow** [comment]
Hey, @KhalidAlt . Is there any chance you could add all the categories (listed [here](https://www.kaggle.com/datasets/rmisra/news-category-dataset))  to the `answer_choices` fields so that rank/accuracy scoring could easily be performed? Thanks!

**Comment by KhalidAlt** [comment]
I add  choices_in_prompt as @jon-tow suggested. However, the categories is listed as string instead of labels and I can not use {{answer_choice[category]}} as suggested in the prompts guide. Therefore, I just wrote {{category}} as the target. Does it still work correctly with rank/accuracy regardless of this issue or should I edit the dataset to include  categories as labels instead of just string.

**Comment by thinkzink** [comment]
I think the following could be reworded to make the description more separable from the surrounding text but I think the others look fine: 'Given the following article headline: "{{headline}}" and the following
      short passage "{{short_description}}" What is the category of this news report?
      ||| {{category}}'
      
For consistency it would be important to edit the dataset for that {{answer_choice[category]}} ability in including categories as labels.

**Review by thinkzink — APPROVED** [review]

**Comment by cjlovering** [comment]
@thinkzink @KhalidAlt It looks like the changes are approved. Should this be merged?

**Comment by KhalidAlt** [comment]
> @thinkzink @KhalidAlt It looks like the changes are approved. Should this be merged?

Yes I think so.

**Review by thinkzink — APPROVED** [review]

File: promptsource/templates/khalidalt/HuffPost/templates.yaml
Comment (awebson) [inline promptsource/templates/khalidalt/HuffPost/templates.yaml:11]: This is ungrammatical. How about "Given these headline and description, what category can you infer this piece of news belongs to?"

File: promptsource/templates/khalidalt/HuffPost/templates.yaml
Comment (awebson) [inline promptsource/templates/khalidalt/HuffPost/templates.yaml:88]:
```suggestion
    jinja: "Given the following article headline and short passage of the article,\
      \ what is the category of this news report? \nArticle headline: {{headline}}\n\
```

File: promptsource/templates/khalidalt/HuffPost/templates.yaml
Comment (awebson) [inline promptsource/templates/khalidalt/HuffPost/templates.yaml:29]: Why doesn't this prompt also use the short description? Is the original task intended to just use the headline or both?

**Review by awebson — CHANGES_REQUESTED** [review]
Thanks for the PR! A few issues left: 

1. None of your prompts have the answer choices in the input, so (a) models or even humans will never know to answer "impact" or "divorce" as a valid news category. (b) This may be mitigated by rank classification of the answer choices, but it's far from guaranteed. At the very lest, you need to set the `choices_in_prompts` accordingly. 
2. Please lowercase or title case your answer choices. Most models are case-sensitive now and are a priori unlikely to generate all caps vocabulary.
3. The classes are imbalanced, so you probably don't really want to report naive accuracy. Fine for now, but please pay attention in you evaluation.
4. What is the file change of `promptsource/templates/tydiqa/primary_task/.templates.yaml.swp`?

File: promptsource/templates/khalidalt/HuffPost/templates.yaml
Comment (KhalidAlt) [reply promptsource/templates/khalidalt/HuffPost/templates.yaml]: > Thanks for the PR! A few issues left:
> 
>     1. None of your prompts have the answer choices in the input, so (a) models or even humans will never know to answer "impact" or "divorce" as a valid news category. (b) This may be mitigated by rank classification of the answer choices, but it's far from guaranteed. At the very lest, you need to set the `choices_in_prompts` accordingly.
> 
>     2. Please lowercase or title case your answer choices. Most models are case-sensitive now and are a priori unlikely to generate all caps vocabulary.
> 
>     3. The classes are imbalanced, so you probably don't really want to report naive accuracy. Fine for now, but please pay attention in you evaluation.
> 
>     4. What is the file change of `promptsource/templates/tydiqa/primary_task/.templates.yaml.swp`?

1- I can add answer choices as list in the input as follow:

`Given the following news headline and short description: 
{{headline}}
{{short_description}}
What is a correct category class from the following list {{answer_choices}} ? ||| {{answer_choices[label]}}`

However, It may seem unnatural to use this style with humans. 

2- Fixed. I titled case all answer choices. 

3- OK

4- Fixed. It was a mistake and I removed the file.

**Review by awebson — APPROVED** [review]
Thank you!

> I can add answer choices as list in the input as follow:
> 
> Given the following news headline and short description: {{headline}} {{short_description}} What is a correct category class from the following list {{answer_choices}} ? ||| {{answer_choices[label]}}
> 
> However, It may seem unnatural to use this style with humans.

I think it's pretty natural to ask people and models something like

> choose from the list of options below:
> - politics
> - sports
> - ...

But I will leave it up to you. In our experience, including the answer choices could either make a huge difference or nothing at all. You should definitely try it both ways in your evaluation. Let me know if you want me to merge this now or wait until you include some prompts with answer choices.

**Comment by KhalidAlt** [comment]
Now it is done! I add answer_choices in 3 of the prompts and kept two without any answer_choices in the prompts. Please review the last commits and merge it if everything is OK.

**Comment by awebson** [comment]
Beautiful. Thanks so much!

**Pull request merged**
